### DISCUSSION TITLE
Missing features?

### DISCUSSION THREAD

**Discussion by ma-galan** [JetBrains/lets-plot#897] [General]
Hi all, first of all congrats for such a wonderful Framework for people like me that love working with grammar of graphics plotting APIs.
I've been checking this framework for a while and I'm missing (or I can't find how to do) the following features that would totally make me move all my future developments to lets-plot:
(1) Is there any way to add more intetactivity to the plots? Pan, zoom in/out, box zooms, selections...
(2) Can I use a second y axis (on the right) independently in the same plot for a different set of variables than the ones used in the left y axis?
(3) Is there a way to make subplots sharing the x-axis?

I want to know if these features are already implemented or there is any plan to do so.

Thanks everyone!

**Comment by alshan** [comment]
Hi @Episkiliski ! Unfortunately, all features you mentioned are missing in current lets-plot.
(3) is in plans: #718

(1), (2) are rather in "research" stage :)  

btw, what would you expect from the "selections" feature?

**Reply by OSuwaidi** [reply]
I really like the idea of feature (1), especially the "selections" feature. 

In Bokeh for example, there is the `LassoSelectTool()` feature which allows you to interactively select points from a plot. And the neat thing is, if a group of points is selected, you can then get their details (features) and display them below the plot (it keeps track of what's been selected then invokes a callback function via the `.on_change()` method)!

That feature is super useful and intuitive, and it would be a perfect addition to my favorite `lets-plot` framework!

**Comment by OSuwaidi** [comment]
Hey @alshan 👋🏼,

I've noticed that plots created with `lets-plot` do not currently support a responsive mode. This slightly cripples UX, because I am generating the plots in a web app, and it's being accessed by users on various devices, making the fixed width and height of plots inconvenient!

Are there any plans to introduce responsiveness to the plots (similar to width/height policy in Bokeh)?

**Reply by alshan** [reply]
Submitted an issue #1025, thanks.

**Reply by alshan** [reply]
Awaiting response: https://github.com/JetBrains/lets-plot/issues/1025#issuecomment-1984581295
